### PR TITLE
Clear output path by default

### PR DESCRIPTION
**Pull request by jorgerangel-msft**
Updates the generator to clear the generated output folder by default.

fixes: https://github.com/microsoft/typespec/issues/3469

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public async Task ExecuteAsync()
 
             if (CodeModelPlugin.Instance.Configuration.ClearOutputFolder)
             {
-                DeleteDirectory(outputPath, _filesToKeep);
+                DeleteDirectory(ParseOutputPath(outputPath), _filesToKeep);
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs:71]: Why do we need to call ParseOutputPath here?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs [hunk 2]
```diff
@@ -98,6 +98,8 @@ internal static string ParseOutputPath(string outputPath)
                 outputPath = Path.Combine(outputPath, "src");
             }
 
+            outputPath = Path.Combine(outputPath, GeneratedFolderName);
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs:101]: Do we really need this method? It seems like it is only used for tests. Can we move it into the test class that needs it and make GeneratedFolderName internal?

Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs:101]: Nevermind, I see that this is the Generated Source output path. Can we rename the method to something like ParseGeneratedSourceOutputPath?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs [hunk 3]
```diff
@@ -24,12 +24,12 @@ public async Task ExecuteAsync()
...
             GeneratedCodeWorkspace.Initialize();
             var outputPath = CodeModelPlugin.Instance.Configuration.OutputDirectory;
-            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", "Generated");
+            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", GeneratedFolderName);
 
             GeneratedCodeWorkspace workspace = await GeneratedCodeWorkspace.Create();
 
             var output = CodeModelPlugin.Instance.OutputLibrary;
-            Directory.CreateDirectory(Path.Combine(outputPath, "src", "Generated", "Models"));
+            Directory.CreateDirectory(Path.Combine(outputPath, "src", GeneratedFolderName, "Models"));
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs:32]: nit: can cache off the generatedSourceOutputPath after calling the method, and then do a Path.Combine on that and "Models".

**Review by JoshLove-msft — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs [hunk 4]
```diff
@@ -24,12 +24,13 @@ public async Task ExecuteAsync()
...
             var outputPath = CodeModelPlugin.Instance.Configuration.OutputDirectory;
-            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", "Generated");
+            var generatedSourceOutputPath = ParseGeneratedSourceOutputPath(outputPath);
+            var generatedTestOutputPath = Path.Combine(outputPath, "..", "..", "tests", GeneratedFolderName);
 
             GeneratedCodeWorkspace workspace = await GeneratedCodeWorkspace.Create();
 
             var output = CodeModelPlugin.Instance.OutputLibrary;
-            Directory.CreateDirectory(Path.Combine(outputPath, "src", "Generated", "Models"));
+            Directory.CreateDirectory(Path.Combine(generatedSourceOutputPath, "Models"));
```
Comment (m-nash) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpGen.cs:33]: If we are creating the directory here https://github.com/microsoft/typespec/pull/3616/files#diff-9bfb4971a041afb747e174ac1e942fb8e1fe1343bee198af96931f0df82fdb2bR85, why is this line needed?
